push-payload/index.js

```diff
@@ -2,20 +2,51 @@ var endpoint;
 var key;
 var authSecret;
 
+function urlBase64ToUint8Array(base64String) {
+  const padding = '='.repeat((4 - base64String.length % 4) % 4);
+  const base64 = (base64String + padding)
+    .replace(/\-/g, '+')
+    .replace(/_/g, '/');
+ 
+  const rawData = window.atob(base64);
+  const outputArray = new Uint8Array(rawData.length);
+ 
+  for (let i = 0; i < rawData.length; ++i) {
+    outputArray[i] = rawData.charCodeAt(i);
+  }
+  return outputArray;
+}
+
 // Register a Service Worker.
 navigator.serviceWorker.register('service-worker.js')
 .then(function(registration) {
   // Use the PushManager to get the user's subscription to the push service.
   return registration.pushManager.getSubscription()
-  .then(function(subscription) {
-    // If a subscription was found, return it.
+  .then(async function(subscription) {
+    // Get the server's public key
+    var response = await fetch('./vapidPublicKey')
+    var vapidPublicKey = await response.text();
+    // Convert the URL safe base64 string to a Uint8Array
+    var convertedVapidKey = urlBase64ToUint8Array(vapidPublicKey);
```

Comment:
You can probably use the base64 string directly, see https://developer.mozilla.org/en-US/docs/Web/API/PushManager/subscribe

Reply:
Doesn't seem to work on Chrome:
`TypeError: Failed to execute 'subscribe' on 'PushManager': The provided value is not of type '(ArrayBuffer or ArrayBufferView)'`

Reply:
OK, so I guess it's not implemented yet in Chrome. Unfortunately I don't think there's a way to share the code between the recipes, so we will have to copy it everywhere.
Please also add a comment explaining why it is needed and that it can be removed once Chrome supports passing a base64 string.

Reply:
Related [Chrome bug](https://bugs.chromium.org/p/chromium/issues/detail?id=802280)